### PR TITLE
Bug 1172349 - [Settings] Convert SimPin panel switch to use gaia-switch r=gasolin

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1172349

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a8217909159a6387d361fc639652bd5ab8b45229

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c73dab71fd3e141ed3df552433f13cba0c8e76e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5e3598610e34cbf263b452ca5b80054ec9f267fe

File: apps/settings/js/panels/simpin/simpin.js [hunk 1]
```diff
@@ -34,12 +36,11 @@ define(function(require) {
...
-            <input type="checkbox" data-ignore data-sim-index="${simIndex}"
-              data-type="checkSimPin"/>
-            <span data-l10n-id="simPinWithIndex"
-              data-l10n-args="${simPinArgs}"></span>
-        </label>
+          <gaia-switch data-ignore data-sim-index="${simIndex}"
+              data-type="checkSimPin">
+            <label data-l10n-id="simPinWithIndex"
+              data-l10n-args="${simPinArgs}"></label>
+        </gaia-switch>
```
Comment (gasolin) [inline apps/settings/js/panels/simpin/simpin.js]: wrong indent

File: apps/settings/js/panels/simpin/simpin.js [hunk 2]
```diff
@@ -126,15 +127,17 @@ define(function(require) {
       cardIndex = parseInt(cardIndex, 10);
 
       switch (type) {
-        case 'checkSimPin':
```
Comment (gasolin) [inline apps/settings/js/panels/simpin/simpin.js]: why not keep this switch condition, remove `simPinSwitched` and handle with `this._elements.simPinContainer.addEventListener('change', this)` ?

File: apps/settings/js/panels/simpin/simpin.js [hunk 3]
```diff
@@ -89,14 +90,14 @@ define(function(require) {
...
-          '.simpin-enabled-' + cardIndex + ' input');
+          '.simpin-enabled-' + cardIndex + ' gaia-switch');
 
       var isSimAvailable = icc && icc.cardState && icc.cardState !== 'unknown';
 
       // when fugu is in airplane mode, icc.cardState will not be changed ...
       // in this way, we have to use isAirplaneMode to check this situation
       if (!isSimAvailable || this.isAirplaneMode) {
-        simPinCheckbox.disabled = true;
+        simPinCheckbox.setAttribute('disabled', true);
```
Comment (gasolin) [inline apps/settings/js/panels/simpin/simpin.js]: seems it's frequently used, why we do not use `Object.defineProperty( proto, 'disabled' ...` to define this property in `gaia-switch` element?

**Pull request closed without merging**